### PR TITLE
Block user

### PR DESCRIPTION
**Pull request by istzen**
This is the flutter PR for the blocking of a User. The only thing that happens here is that a user that is blocked can't sign in anymore. The Important part of this functionality will be in the corresponding admin PR, which can be found [here](https://github.com/MOTIS-Mitfahr-App/react-web/pull/6).

### CODE REVIEW

File: lib/l10n/app_de.arb [hunk 1]
```diff
@@ -19,6 +19,8 @@
   "before": "Vorher",
   "after": "Nachher",
   "failureSnackBar": "Es ist ein Fehler aufgetreten.",
+  "blockedSnackBar": "Dein Account wurde geblockt, bitte kontaktiere uns für mehr Information",
```
Comment (fynsta) [inline lib/l10n/app_de.arb:22]: Informationen

File: lib/main.dart [hunk 2]
```diff
@@ -83,6 +84,16 @@ class AuthAppState extends State<AuthApp> {
         final AuthChangeEvent event = data.event;
         final Session? session = data.session;
         await supabaseManager.reloadCurrentProfile();
+        if (event == AuthChangeEvent.signedIn) {
+          final bool? isBlocked = await supabaseManager.supabaseClient
+              .rpc('is_blocked', params: <String, dynamic>{'user_id': data.session!.user.id}) as bool?;
+          if (isBlocked != null && isBlocked) {
```
Comment (fynsta) [inline lib/main.dart:90]:
```suggestion
          if (isBlocked ?? false) {
```

File: test/widgets/main_test.dart [hunk 3]
```diff
@@ -151,6 +153,31 @@ void main() {
       expect(find.byType(MainApp), findsOneWidget);
     });
 
+    testWidgets('It shows WelcomePage after signing in when user is blocked', (WidgetTester tester) async {
```
Comment (fynsta) [inline test/widgets/main_test.dart:156]: "It logs user out again after..."

**Comment by fynsta** [comment]
Is there no way to block it straight on login? Login - Logout feels dirty?

**Comment by istzen** [comment]
> Is there no way to block it straight on login? Login - Logout feels dirty?

the only way would be to write a new function for the signup with checks whether a user is blocked before the sign in.

**Comment by fynsta** [comment]
As we said, in the future it would be great if we could somehow make sure that the user does not do anything while he is not logged out yet. But of course I have no idea how to do that, and we don't have any time anymore.

**Review by fynsta — APPROVED** [review]

**Pull request merged**